gateway/crates/gateway-binary/tests/telemetry/mod.rs

```diff
@@ -94,3 +96,71 @@ fn with_otel() {
         assert!(count > 0);
     });
 }
+
+#[test]
+fn with_otel_reload_tracing() {
+    let service_name = format!("service-{}", rand::random::<u128>());
+    let config = &formatdoc! {r#"
+        [telemetry]
+        service_name = "{service_name}"
+
+        [telemetry.tracing]
+        enabled = true
+        sampling = 1
+
+        [telemetry.tracing.exporters.otlp]
+        enabled = true
+        endpoint = "http://localhost:4317"
+        protocol = "grpc"
+
+        [telemetry.tracing.exporters.otlp.batch_export]
+        scheduled_delay = 1
+        max_export_batch_size = 1
+    "#};
+
+    let schema = load_schema("big");
+
+    let query = indoc! {r#"
+        { __typename }
+    "#};
+
+    with_hybrid_server(config, "test_graph", &schema, |client| async move {
+        let result: serde_json::Value = client.gql(query).send().await;
+        serde_json::to_string_pretty(&result).unwrap();
+
+        let client = clickhouse::Client::default()
+            .with_url("http://localhost:8123")
+            .with_user("default")
+            .with_database("otel");
+
+        #[derive(clickhouse::Row, Deserialize)]
+        struct CountRow {
+            count: u32,
+        }
+
+        let uplink_mock = UplinkResponseMock::mock("");
```

Comment:
instead of creating another mock response here, I would just pass the `UplinkResponse` as an argument to the closure.